datasources.go

```diff
@@ -36,6 +42,16 @@ func (api API) CreateDataSource(name string) (*DataSource, error) {
 	return ds, err
 }
 
+// CreateDataSource2 creates an API Data Source in ChartMogul.
+// * Allows other parameters than just the name.
+//
+// See https://dev.chartmogul.com/v1.0/reference#data-sources
+func (api API) CreateDataSource2(dataSource *DataSource) (*DataSource, error) {
```

Comment:
What do you propose instead of `CreateDataSource2`? It's completely the same function, except it allows extra attribute, but I didn't want to break the API.
actually it accepts a `struct` instead of a `string`.

Similar problem is with `ListDataSources2`
Previous version doesn't accept argument, this one accepts a `struct`.
We could totally misuse variadric arguments to make it backwards compatible and no new functions, but... I find that quite dirty, since Go is supposed to be strictly typed.